treegp/gp_interp.py

```diff
@@ -172,11 +173,14 @@ def return_gp_predict(self, y, X1, X2, kernel, y_err, return_cov=False):
         :param y_err:  Error of y. (n_samples)
         """
         HT = kernel.__call__(X2, Y=X1)
-        K = kernel.__call__(X1) + np.eye(len(y)) * y_err**2
-        factor = (cholesky(K, overwrite_a=True, lower=False), False)
-        alpha = cho_solve(factor, y, overwrite_b=False)
-        y_predict = np.dot(HT, alpha.reshape((len(alpha), 1))).T[0]
+        if self._alpha is None:
+            K = kernel.__call__(X1) + np.eye(len(y)) * y_err**2
+            factor = (cholesky(K, overwrite_a=True, lower=False), False)
+            self._alpha = cho_solve(factor, y, overwrite_b=False)
+        y_predict = np.dot(HT, self._alpha.reshape((len(self._alpha), 1))).T[0]
         if return_cov:
+            if self._alpha is not None:
```

Comment:
I don't think it's possible for `self._alpha` to be None here, so this check won't help avoid recomputing K. I guess you could set `K = None` before line 176, then do `if K is None:...` here.